### ISSUE TITLE
Re-ordering form fields on large forms produces erratic scrolling behaviour

### ISSUE THREAD

**Issue by harryfleetwood-cti** [umbraco/Umbraco.Forms.Issues#1243]
Our solution uses Umbraco Forms to build and manage forms that are in excess of 100 field types.

It's becoming increasingly frustrating for both our team and the client to manage these forms when re-ordering fields as the form builder does not scroll back to the previous position, or any logical position, but rather a random place on the form builder that then takes time to find the position we were once in.

Reproduction
------------

Create a form with a large amount of pages and field types

Scroll down to a page and click "Reorder"

Move a field type up a few pages

Click "I am done re-ordering"

The page will place the user at a random location further up the page, making it tedious for them to find the page they were originally editing

Screen recording captured here:

[Screen recording](https://gyazo.com/022e2ef3c551d000ffcc6d05fcb4c42a)

In the screen recording, I am editing page 14, and move a field type up to page 7. When I click "I am done re-ordering", I am now position on page 3, and need to scroll far back down the page to where I originally was.

---
_This item has been added to our backlog AB#42185_

**Comment by AndyButland** [comment]
Would it be possible to export and share that form please @harryfleetwood-cti?  It looks like a test one so that's hopefully OK.  Just would be handy to help with replicating the issue without having to manually create such a large form.  Thanks if so.

**Comment by harryfleetwood-cti** [comment]
@AndyButland 

Export of the form here, thanks

https://drive.google.com/file/d/1NANV5IDyo97gvObdU5UKU6p54i-6Svph/view

**Comment by AndyButland** [comment]
I've added a couple of updates to try to handle situations like this with very large, multi-page forms.

Firstly I've added some scroll position tracking, such that when you switch between the definition and ordering view, you'll start from the same presented page.    Seems to work as expected though it proved a little tricky due to the scrollable containers and sticky headers we have.

Secondly I've added a jump to page dropdown in the header, so an editor can go quickly to a specific page number:

![image](https://github.com/umbraco/Umbraco.Forms.Issues/assets/1993459/07d1938e-5123-4c2b-bc05-42eff9466f75)

This will be in the 13.2.0 release candidate due out tomorrow.
